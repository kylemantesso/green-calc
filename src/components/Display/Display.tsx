import styled from "styled-components/native";
import React from "react";
import {Colours} from "../../style/Colours";

const DisplayText = styled.Text`
  font-size: 58px;
  padding: 16px;
`;

const DisplayContainer = styled.View`
  flex: 1;
  flex-direction: column;
  background-color: ${Colours.LIGHT_GREEN};
  align-items: flex-end;
  justify-content: flex-end;
`;

export const Display = ({ digits }: { digits: string }) => (
  <DisplayContainer>
    <DisplayText>{digits}</DisplayText>
  </DisplayContainer>
);
