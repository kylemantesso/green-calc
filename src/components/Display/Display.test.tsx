import React from "react";

import renderer from "react-test-renderer";
import { Display } from "./Display";

test("renders correctly", () => {
  const digits = "123";
  const tree = renderer.create(<Display digits={digits}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
