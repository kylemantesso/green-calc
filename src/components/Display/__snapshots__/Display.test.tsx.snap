// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#a8ff3e",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 58,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
      ]
    }
  >
    123
  </Text>
</View>
`;
